File: src/main/java/leetcode/problems/q1567/SolutionFail.java
package leetcode.problems.q1567;

class SolutionFail {

    boolean[] dp;

    public int getMaxLen(int[] nums) {
        int answer = 0;

        for (int i = 0; i < nums.length; i++) {
            dp = new boolean[nums.length];
            dp[i] = nums[i] > 0 ? true : false;
            if (nums[i] == 0) {
                continue;
            }
            if (dp[i]) {
                answer = Math.max(answer, 1);
            }

            for (int j = i + 1; j < nums.length; j++) {
                boolean value = nums[j] > 0 ? true : false;
                dp[j] = !(dp[j - 1] ^ value);

                if (nums[j] == 0) {
                    break;
                }
                if (dp[j]) {
                    answer = Math.max(answer, j - i + 1);
                }
            }
        }

        return answer;
    }
}

File: src/main/kotlin/leetcode/q2444/SolutionFail.kt
package leetcode.q2444

import java.util.PriorityQueue
import kotlin.math.max


class SolutionFail {
    fun countSubarrays(nums: IntArray, minK: Int, maxK: Int): Long {
        var answer = 0L

        val minIndexPq = PriorityQueue<Int>()
        val maxIndexPq = PriorityQueue<Int>()


        var tail = 0

        for ((index, num) in nums.withIndex()) {

            if (num < minK || num > maxK) {

                while (tail <= num && minIndexPq.isNotEmpty() && maxIndexPq.isNotEmpty()) {

                    val minIndex = minIndexPq.peek()
                    val maxIndex = maxIndexPq.peek()

                    if (minK == maxK) {
                        answer += index - tail
                    } else {
                        answer += (index - max(minIndex, maxIndex))
                    }

                    if (minIndex == tail) {
                        minIndexPq.poll()
                    }

                    if (maxIndex == tail) {
                        maxIndexPq.poll()
                    }

                    tail++
                }

                minIndexPq.clear()
                maxIndexPq.clear()
                tail = index + 1
            }

            if (num == minK) {
                minIndexPq.add(index)
            }

            if (num == maxK) {
                maxIndexPq.add(index)
            }
        }

        while (tail < nums.size && minIndexPq.isNotEmpty() && maxIndexPq.isNotEmpty()) {

            val minIndex = minIndexPq.peek()
            val maxIndex = maxIndexPq.peek()

            if (minK == maxK) {
                answer += nums.size - tail
            } else {
                answer += (nums.size - max(minIndex, maxIndex))
            }

            if (minIndex == tail) {
                minIndexPq.poll()
            }

            if (maxIndex == tail) {
                maxIndexPq.poll()
            }

            tail++
        }

        return answer
    }
}